src/models/thread.ts

```diff
@@ -192,6 +192,18 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
         }
     };
 
+    private onTimelineEvent = (
+        event: MatrixEvent,
+        room: Room | undefined,
+        toStartOfTimeline: boolean | undefined,
+    ): void => {
+        // Add a synthesized receipt to all live events added to the timeline
+        if (!toStartOfTimeline) {
+            room!.addLocalEchoReceipt(event.getSender()!, event, ReceiptType.Read);
```

Comment:
Won't this possibly clobber a main room timeline RR given that `threadId` prefers threads and falls back to `MAIN_ROOM_TIMLINE`

Reply:
I'm not sure I follow the scenario you're talking about, nor what the risk would be?

Reply:
that a RR the thread root in the main timeline will vanish as soon as it's sender sends a message into the thread and a synthetic RR is generated with a `thread_id` which isn't `MAIN_ROOM_TIMELINE`